sds/src/secondary_validation/romanian_personal_numeric_code.rs

```diff
@@ -0,0 +1,79 @@
+use crate::secondary_validation::Validator;
+pub struct RomanianPersonalNumericCode;
+
+const CHECKSUM_WEIGHTS: &[u32] = &[2, 7, 9, 1, 4, 6, 3, 5, 8, 2, 7, 9];
+
+impl Validator for RomanianPersonalNumericCode {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        if !regex_match.is_ascii() {
+            return false;
+        }
+        for c in regex_match.chars() {
+            if !c.is_ascii_digit() {
+                return false;
+            }
+        }
```

Comment:
We could make this checksum a bit more future proof by removing the separators instead of declaring the checksum as non valid if there is any separator (the regex should check the number of digits in any case.)

This way, filter_map could he used here
```rust
        let mut digits = regex_match.chars().filter_map(|c| c.to_digit(10));
```